functions/src/utils.ts

```diff
@@ -103,3 +100,41 @@ export function isValidUrl(urlString: string | undefined): boolean {
   }
   return url.protocol === "http:" || url.protocol === "https:";
 }
+
+/** Verify a Firebase ID token is valid.
+ *
+ * @param token Firebase ID token to verify.
+ * @returns True if the token is valid, throws an error otherwise.
+ */
+export async function verifyIdToken(token: string | undefined) {
+  if (!token) {
+    throw new ShareLinkError(ShareLinkErrorCode.INVALID_TOKEN);
+  }
+  return admin
+    .auth()
+    .verifyIdToken(token)
+    .then(() => true)
+    .catch((error) => {
+      console.error(error);
```

Comment:
Can you verify this is intentional?

Reply:
It is, but we can probably take it out. Was logging the internal error as throwing a share link error without anything else loses the internal message (which might be worth revisiting in the ShareLinkError structure/setup)

Reply:
Feel free to keep it if it's useful!